watcher_prover/Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM elixir:1.14.5-otp-25
+
+# Copy watcher_prover inside docker
+WORKDIR /watcher
+COPY . /watcher
```

Comment:
As a best practice on Docker in general, you want to `COPY` the code as late as possible (for example, here it would be after rust installation). If you don't to this, every time you modify some code, you have to re-build the entire docker image again, instead of caching steps like apt and rust installs.

Reply:
More info: https://docs.docker.com/build/cache/

Reply:
Fixed on 2ae6bdd!